source/Examples/SmartCache/SmartCache.ExperimentSession/SmartCacheExperimentSession.cs

```diff
@@ -86,17 +86,17 @@ public SmartCacheExperimentSession(MlosContext mlosContext)
             //
             Hypergrid cacheSearchSpace = new Hypergrid(
                 name: "smart_cache_config",
-                dimension: new CategoricalDimension("cache_implementation", CacheEvictionPolicy.LeastRecentlyUsed, CacheEvictionPolicy.MostRecentlyUsed))
+                dimension: new CategoricalDimension("cache_implementation", (int)CacheEvictionPolicy.LeastRecentlyUsed, (int)CacheEvictionPolicy.MostRecentlyUsed))

```

Comment:
do we really need a cast ?

Reply:
Without the cast the object is passed as an enum and is not converted to an int. CategoricalDimensions as per the proto file are only allowed to hold PrimitiveValue s. 

Do you have another idea?

Reply:
Smartcache uses a string categorical dimension rather than int
11065c0

Is this better?